pkg/controller.v1/tensorflow/pod_test.go

```diff
@@ -274,6 +274,51 @@ var _ = Describe("TFJob controller", func() {
 		})
 	})
 
+	Context("Test Unretryable Exit Code", func() {
+		It("should set the job status to Failed", func() {
+			By("Creating TFJob \"test-noretry-exit-code\" with 1 worker only")
+			ctx := context.Background()
+
+			tfJob := tftestutil.NewTFJob(1, 0)
+			tfJob.SetName("test-noretry-exit-code")
+			tfJob.SetUID(uuid.NewUUID())
+			tfJob.Spec.TFReplicaSpecs[kubeflowv1.TFJobReplicaTypeWorker].RestartPolicy = kubeflowv1.RestartPolicyExitCode
+			Expect(testK8sClient.Create(ctx, tfJob)).Should(Succeed())
+
+			_ = reconciler.ReconcileJobs(tfJob, tfJob.Spec.TFReplicaSpecs, tfJob.Status, &tfJob.Spec.RunPolicy)
+
+			created := &corev1.Pod{}
+			key := types.NamespacedName{Namespace: metav1.NamespaceDefault, Name: "test-noretry-exit-code-worker-0"}
+			Expect(testK8sClient.Get(ctx, key, created)).Should(Succeed())
+			created.Status.Phase = corev1.PodFailed
+			created.Status.ContainerStatuses = append(created.Status.ContainerStatuses, corev1.ContainerStatus{
+				Name: kubeflowv1.TFJobDefaultContainerName,
+				State: corev1.ContainerState{
+					Terminated: &corev1.ContainerStateTerminated{
+						ExitCode: 1,
+					},
+				},
+			})
+			Expect(testK8sClient.Status().Update(ctx, created))
+
+			_ = reconciler.ReconcileJobs(tfJob, tfJob.Spec.TFReplicaSpecs, tfJob.Status, &tfJob.Spec.RunPolicy)
+
+			Eventually(func() bool {
+				updatedJob := &kubeflowv1.TFJob{}
+				err := testK8sClient.Get(ctx, types.NamespacedName{Name: tfJob.GetName(), Namespace: metav1.NamespaceDefault}, updatedJob)
+				if err != nil {
+					return false
+				}
+				for _, condition := range updatedJob.Status.Conditions {
+					if condition.Type == kubeflowv1.JobFailed && condition.Status == corev1.ConditionTrue {
+						return true
+					}
+				}
+				return false
+			}, testutil.Timeout, testutil.Interval).Should(BeTrue(), "TFJob should be in Failed state")
```

Comment:
```suggestion
			Eventually(func(g Gomega) {
				updatedJob := &kubeflowv1.TFJob{}
				g.Expect(testK8sClient.Get(ctx, types.NamespacedName{Name: tfJob.GetName(), Namespace: metav1.NamespaceDefault}, updatedJob)).Should(Succeeded())
				g.Expect(updatedJob.Status.Conditions).Should(ContainElements(
					BeComparableTo(metav1.Condition{
						Type: kubeflowv1.JobFailed,
						Status: corev1.ConditionTrue,
					}, cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime", "Reason", "Message", "ObservedGeneration"), "TFJob should be in Failed state"),
				))
			}, testutil.Timeout, testutil.Interval).Should(Succeeded())
```

Could you improve debuggability like this?

Reply:
Done!